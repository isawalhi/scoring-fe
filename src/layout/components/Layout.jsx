import React from 'react';
import { Link } from 'react-router-dom';
import { Layout as AntdLayout, Menu, Spin } from 'antd';
import styles from './Layout.module.css';
import getRoutes from '../../routes';

const { Header, Content, Footer } = AntdLayout;

const Layout = ({ children, showRequestLoader }) => {
  return (
    <AntdLayout className={styles.appContainer}>
      <Header>
        <div className={styles.pageLogo} />
        <Menu theme="dark" mode="horizontal" defaultSelectedKeys={['1']}>
          <Menu.Item key={getRoutes.gamesRoute.path}>
            <Link to={getRoutes.gamesRoute.path}>All Times</Link>
          </Menu.Item>
          <Menu.Item key={getRoutes.addGameRoute.path}>
            <Link to={getRoutes.addGameRoute.path}>Add new game</Link>
          </Menu.Item>
        </Menu>
      </Header>
      {showRequestLoader && <Spin size="large" />}
      <Content className={styles.pageContent}>{children}</Content>
      <Footer theme="dark" className={styles.pageFooter}>
        Scopely Â©2020 Created by Sawalhi
      </Footer>
    </AntdLayout>
  );
};

export default Layout;
