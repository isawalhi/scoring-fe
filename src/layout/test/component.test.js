import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Layout from '../components/Layout';

it('renders', () => {
  const wrapper = shallow(<Layout />);
  expect(toJson(wrapper)).toMatchSnapshot();
});
