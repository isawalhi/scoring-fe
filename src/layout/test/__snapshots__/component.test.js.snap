// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<Layout
  className="appContainer"
>
  <Header>
    <div
      className="pageLogo"
    />
    <Menu
      defaultSelectedKeys={
        Array [
          "1",
        ]
      }
      mode="horizontal"
      theme="dark"
    >
      <MenuItem
        key="/all-time"
      >
        <Link
          to="/all-time"
        >
          All Times
        </Link>
      </MenuItem>
      <MenuItem
        key="/add-game"
      >
        <Link
          to="/add-game"
        >
          Add new game
        </Link>
      </MenuItem>
    </Menu>
  </Header>
  <Content
    className="pageContent"
  />
  <Footer
    className="pageFooter"
    theme="dark"
  >
    Scopely Â©2020 Created by Sawalhi
  </Footer>
</Layout>
`;
